### PR TITLE
fix: broken link in `install` script

### PR DESCRIPTION
**Pull request by intelliot**
The address led to a 404 Page Not Found. This fixes the URL.

### CODE REVIEW

File: hummingbot/VERSION [hunk 1]
```diff
@@ -1 +1 @@
-dev-1.29.0
+2.0.0
```
Comment (cardosofede) [inline hummingbot/VERSION:1]: @rapcmia we need to update the dev version right?

File: install [hunk 2]
```diff
@@ -12,7 +12,7 @@ fi
 
 if [ "${CONDA_EXE}_" == "_" ]; then
     echo "Please install Anaconda w/ Python 3.7+ first"
-    echo "See: https://www.anaconda.com/distribution/"
+    echo "See: https://www.anaconda.com/download"
```
Comment (cardosofede) [inline install:15]: the changes are okey to me

File: setup.py [hunk 3]
```diff
@@ -33,7 +33,7 @@ def build_extensions(self):
 
 def main():
     cpu_count = os.cpu_count() or 8
-    version = "20240531"
+    version = "20240703"
```
Comment (cardosofede) [inline setup.py:36]: @rapcmia same here

**Review by cardosofede — CHANGES_REQUESTED** [review]
The PR is okey to merge but we need to make sure that the versions are updated.
@intelliot here are the contribution guidelines: https://hummingbot.org/developers/contributions/
I changed the base branch to development since all the contributions passes through it

File: hummingbot/VERSION (hunk 1 above)
Comment (nikspz) [reply hummingbot/VERSION:1]: Yes should be dev-2.0.1 for now
https://github.com/hummingbot/hummingbot/blob/development/hummingbot/VERSION

**Review by nikspz — APPROVED** [review]
Checked links, LGTM

**Pull request merged**